src/main/java/org/embulk/output/kintone/KintonePageOutput.java

```diff
@@ -248,22 +237,29 @@ private void upsert(ArrayList<Record> records, ArrayList<UpdateKey> updateKeys)
         updateRecords.clear();
       }
     }
-    if (insertRecords.size() > 0) {
+    if (!insertRecords.isEmpty()) {
       client.record().addRecords(task.getAppId(), insertRecords);
     }
-    if (updateRecords.size() > 0) {
+    if (!updateRecords.isEmpty()) {
       client.record().updateRecords(task.getAppId(), updateRecords);
     }
   }
 
   private List<Record> getExistingRecordsByUpdateKey(ArrayList<UpdateKey> updateKeys) {
-    String fieldCode = updateKeys.get(0).getField();
+    String fieldCode =
+        updateKeys.stream()
+            .map(UpdateKey::getField)
+            .filter(Objects::nonNull)
+            .findFirst()
+            .orElse(null);
```

Comment:
Originally, field code was always getting from the first record. However, field code could be null if the input record does not have update key value. So I fixed to get the first non-null field code.